File: test_route.py
import requests
import json

url = "http://localhost:5050/api/route"
data = {
    "source": "torres-phone",
    "target": "cmd-hq", 
    "scenario": "earthquake"
}

response = requests.post(url, json=data)
print(f"Status: {response.status_code}")
print(f"Response: {json.dumps(response.json(), indent=2)}")

File: test_route.js
const http = require('http');

const data = JSON.stringify({
  source: 'torres-phone',
  target: 'cmd-hq',
  scenario: 'earthquake'
});

const options = {
  hostname: 'localhost',
  port: 5050,
  path: '/api/route',
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'Content-Length': data.length
  }
};

const req = http.request(options, (res) => {
  let body = '';
  res.on('data', (chunk) => body += chunk);
  res.on('end', () => {
    console.log('Status:', res.statusCode);
    console.log('Response:', body);
  });
});

req.on('error', (e) => console.error('Error:', e.message));
req.write(data);
req.end();
